lib/tasks/migrations/migrations.rake

```diff
@@ -1,27 +1,33 @@
 # frozen_string_literal: true
 
 namespace :migrations do
+  DEFAULT_ROLES_MAP = { "admin" => "Administrator", "user" => "User" }.freeze
   COMMON = {
-    headers: { "Content-Type" => "application/json" }
+    headers: { "Content-Type" => "application/json" },
+    batch_size: 1000,
   }.freeze
 
   desc "Migrates v2 resources to v3"
   task :roles, [] => :environment do |_task, _args|
     has_encountred_issue = 0
 
-    Role.select(:id, :name).where.not(name: Role::RESERVED_ROLE_NAMES).each do |r|
-      params = { role: { name: r.name } }
-      response = Net::HTTP.post(uri('roles'), payload(params), COMMON[:headers])
-
-      case response
-      when Net::HTTPCreated
-        puts green "Succesfully migrated Role:"
-        puts cyan "  Name: #{r.name}"
-      else
-        puts red "Unable to migrate Role:"
-        puts red "  Name: #{r.name}"
-        has_encountred_issue = 1 # At least one of the migrations failed.
-      end
+    Role.select(:id, :name)
+        .where.not(name: Role::RESERVED_ROLE_NAMES)
+        .find_each(batch_size: COMMON[:batch_size]) do |r|
```

Comment:
I think you can do `in_batches_of` - I would also recommend lowering the size to 50 or 100

Reply:
Please refer to #4010

Reply:
> I think you can do `in_batches_of` - I would also recommend lowering the size to 50 or 100

`find_in_batches_of` will make the same query as `find_each` but instead of instantiating a model for `each` record in the result, it will instead (like `all`) instantiate all of the loaded raw records and yield that to given block as an array,
Since, that isn't (as I'm aware of) will optimize the queries but simply eager load the models and cause us to further iterate the given array I used `find_each` instead,

The batch size is the defaulted value from the lib writers, that indicated me that a 1000 record is acceptable to be loaded into memory besides `find_each` will not have to create a 1000 model instance and probably use only one for each iteration (only assumption),

Do you wanna default that to a 100?
To note, that will cause more queries to fire,